src/features/world/ui/beach/treasure_shop/TreasureShopCraft.tsx

```diff
@@ -25,26 +25,29 @@ export const TreasureShopCraft: React.FC = () => {
       context: { state },
     },
   ] = useActor(gameService);
+  const inventory = state.inventory;
+
   const lessIngredients = () =>
     getKeys(selected.ingredients).some((name) =>
-      selected.ingredients[name]?.greaterThan(state.inventory[name] || 0),
+      selected.ingredients[name]?.greaterThan(inventory[name] || 0),
     );
-  const isAlreadyCrafted =
-    state.inventory[selectedName]?.greaterThanOrEqualTo(1);
+  const isAlreadyCrafted = inventory[selectedName]?.greaterThanOrEqualTo(1);
```

Comment:
I would suggest letting players craft multiple.

This will help burn more Sand (which they will keep accumulating)

Reply:
I'll still need this const for Treasure Map, I'll come up with a different solution for the decorations